packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -429,6 +429,9 @@ function updateMemoComponent(
     child.ref = workInProgress.ref;
     child.return = workInProgress;
     workInProgress.child = child;
+    if (__DEV__) {
+      child._debugOwner = workInProgress;
```

Comment:
Can you tell me why this change was necessary? `_debugOwner` is set automatically for all element types.

The newly added test (from this PR) passes for me in master without this change.